tools/lock_inspect/evm_connection.rb

```diff
@@ -0,0 +1,141 @@
+#!/usr/bin/env ruby
+
+require 'pg'
+require 'yaml'
+require 'optparse'
+
+def error_exit(e, conn, error_message = '', exit_code = 1)
+  puts e.message
+  puts error_message unless error_message.empty?
+  conn.close if conn
+  exit(exit_code)
+end
+
+def process_app_name(app_name)
+  if app_name.end_with?('..')
+    raise "The application name for MIQ worker: {#app_name} is truncated"
+  end
+  items = app_name.split(/[\[\]\s,]+/)
```

Comment:
We can change how we set the application_name to make this easier to parse.

We need to make sure the name is done in order of importance anyway so it might be helpful to make it easier to parse anyway.

For the server, the most important values from left to right:
spid, compressed_id, class name, zone_compressed_id, zone_name

For the workers:
spid, compressed_id, server_compress_id, class name, zone_compressed_id, zone_name